node/src/components/network.rs

```diff
@@ -104,10 +106,14 @@ enum ConnectionState {
 }
 
 #[derive(DataSize)]
-pub(crate) struct Network<REv, P> {
+pub struct Network<REv, P> {
     our_id: NodeId,
     #[data_size(skip)]
     peers: HashMap<NodeId, ConnectedPoint>,
+    /// A set of peers whose address we currently know. Kept in-sync with the internal Kademlia
+    /// routing table.
+    #[data_size(skip)]
+    seen_peers: HashSet<PeerId>,
```

Comment:
This could use more documentation.  I'm not seeing why we need this and also the `peers` member.

Reply:
`peers` tracks connections, which can be arbitrarily opened and closed by libp2p. `seen_peers` tracks peers we know about, regardless if we have ever been or are connected to them. The latter is used by the test to check if node discovery works. This would also be a worthwhile metric to expose.

Reply:
Could you add the critical distinction to the comment then please?  (As in, we may never have been connected to these peers, whereas `peers` represents the currently-connected list of peers)